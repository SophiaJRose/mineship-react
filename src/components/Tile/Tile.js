import * as React from "react"
import tile_unrevealed from "../../images/tile_unrevealed.png"
import tile_mine from "../../images/tile_mine.png"
import tile_0 from "../../images/tile_0.png"

function Tile(props) {
	let images = {
		Unrevealed: tile_unrevealed,
		Mine: tile_mine,
		0: tile_0
	}

	return (
		<button className="tile" onClick={props.onClick}>
			<img src={images[props.image]} alt={props.image}></img>
		</button>
	);
}
Tile.defaultProps = {
	image: "Unrevealed"
}

export { Tile }