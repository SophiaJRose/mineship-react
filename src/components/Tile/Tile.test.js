import React from "react"
import { render, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { Tile } from "./Tile"
import tile_unrevealed from "../../images/tile_unrevealed.png"

describe("<Tile /> tests", () => {
	it("should render as button with image passed to it", () => {
		const { container } = render(<Tile image="tile_unrevealed"/>)
		expect(container.firstChild).toMatchSnapshot()
		expect(container.firstChild).toHaveClass("tile")
		expect(container.firstChild.firstChild).toHaveAttribute("src", tile_unrevealed)
	})

	it("should call the click callback handler", () => {
		// Arrange
		const onClick = jest.fn()
		const { container } = render(<Tile onClick={onClick} />)
		// Act
		fireEvent.click(container.firstChild)
		// Assert
		expect(onClick).toHaveBeenCalledTimes(1)
	})
})