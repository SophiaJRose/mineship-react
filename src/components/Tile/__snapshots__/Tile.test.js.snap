// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> tests should render as button with image tile_unrevealed.png 1`] = `
<button
  class="tile"
>
  <img
    alt="Unrevealed"
    src="tile_unrevealed.png"
  />
</button>
`;
